### DISCUSSION TITLE
Psycopg2 in pydroid3

### DISCUSSION THREAD

**Discussion by doublyU** [psycopg/psycopg2#1572] [General]
I want use psycopg2 in pydroid3, I already installed psycopg2 using termux and copy it working directory of my pydroid3 and is giving me this error below.

Traceback (most recent call last):
  File "/data/user/0/ru.iiec.pydroid3/files/accomp_files/iiec_run/iiec_run.py", line 31, in <module>
    start(fakepyfile,mainpyfile)
  File "/data/user/0/ru.iiec.pydroid3/files/accomp_files/iiec_run/iiec_run.py", line 30, in start
    exec(open(mainpyfile).read(),  __main__.__dict__)
  File "<string>", line 6, in <module>
  File "/storage/emulated/0/python_folder/psycopg2/__init__.py", line 51, in <module>
    from psycopg2._psycopg import (                     # noqa
ModuleNotFoundError: No module named 'psycopg2._psycopg'

[Program finished]

Psycopg2 is not installed by using pydroid3 terminal that is I used termux to install it and copy it the working directory of my pydroid3

**Comment by rohan472000** [comment]
It seems like the installation of `psycopg2` in Termux may not be compatible with the Python environment in Pydroid3.

Instead of manually copying the installed `psycopg2` package from Termux to Pydroid3, I would suggest installing it directly in Pydroid3 using the `pip` package manager. Here are the steps to install psycopg2 in Pydroid3:

1. Open Pydroid3 and open the Terminal from the top menu.

2. In the Terminal, run the following command to install pip if it is not already installed:
          `python -m ensurepip --default-pip`

3. Once `pip` is installed, run the following command to install `psycopg2`:
          `pip install psycopg2`

This should download and install the `psycopg2` package and its dependencies in your Pydroid3 environment.

After installing `psycopg2` with `pip`, you should be able to import it in your Python code without any issues.

**Reply by aarushvarra7-a11y** [reply]
Traceback (most recent call last):
  File "/data/user/0/ru.iiec.pydroid3/files/accomp_files/iiec_run/iiec_run.py", line 31, in <module>
    start(fakepyfile,mainpyfile)
    ~~~~~^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/user/0/ru.iiec.pydroid3/files/accomp_files/iiec_run/iiec_run.py", line 30, in start
    exec(open(mainpyfile).read(),  __main__.__dict__)
    ~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<string>", line 1
    using UnityEngine;
          ^^^^^^^^^^^
SyntaxError: invalid syntax

[Program finished]

**Reply by aarushvarra7-a11y** [reply]
Traceback (most recent call last):
  File "/data/user/0/ru.iiec.pydroid3/files/accomp_files/iiec_run/iiec_run.py", line 31, in <module>
    start(fakepyfile,mainpyfile)
    ~~~~~^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/user/0/ru.iiec.pydroid3/files/accomp_files/iiec_run/iiec_run.py", line 30, in start
    exec(open(mainpyfile).read(),  __main__.__dict__)
    ~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<string>", line 1
    using UnityEngine;
          ^^^^^^^^^^^
SyntaxError: invalid syntax

[Program finished]

**Reply by thailandinifty-gif** [reply]
https://github.com/psycopg/psycopg2/discussions/1572#discussioncomment-17607370

**Comment by doublyU** [comment]
I had done that before dropped message to GitHub but it didn't work I
attached error message that I am getting from terminal to this email. I
really appreciate for respond to me.
I am still looking forward to get reply back from you.

**Comment by doublyU** [comment]
> It seems like the installation of `psycopg2` in Termux may not be compatible with the Python environment in Pydroid3.
> 
> Instead of manually copying the installed `psycopg2` package from Termux to Pydroid3, I would suggest installing it directly in Pydroid3 using the `pip` package manager. Here are the steps to install psycopg2 in Pydroid3:
> 
> 1. Open Pydroid3 and open the Terminal from the top menu.
> 2. In the Terminal, run the following command to install pip if it is not already installed:
>    `python -m ensurepip --default-pip`
> 3. Once `pip` is installed, run the following command to install `psycopg2`:
>    `pip install psycopg2`
> 
> This should download and install the `psycopg2` package and its dependencies in your Pydroid3 environment.
> 
> After installing `psycopg2` with `pip`, you should be able to import it in your Python code without any issues.

I had done that before dropped message to GitHub but it didn't work I attached error message that I am getting from terminal to this email. I really appreciate for respond to me.
I am still looking forward to get reply back from you.

**Comment by doublyU** [comment]
> > It seems like the installation of `psycopg2` in Termux may not be compatible with the Python environment in Pydroid3.
> > Instead of manually copying the installed `psycopg2` package from Termux to Pydroid3, I would suggest installing it directly in Pydroid3 using the `pip` package manager. Here are the steps to install psycopg2 in Pydroid3:
> > 
> > 1. Open Pydroid3 and open the Terminal from the top menu.
> > 2. In the Terminal, run the following command to install pip if it is not already installed:
> >    `python -m ensurepip --default-pip`
> > 3. Once `pip` is installed, run the following command to install `psycopg2`:
> >    `pip install psycopg2`
> > 
> > This should download and install the `psycopg2` package and its dependencies in your Pydroid3 environment.
> > After installing `psycopg2` with `pip`, you should be able to import it in your Python code without any issues.
> 
> I had done that before dropped message to GitHub but it didn't work I attached error message that I am getting from terminal to this email. I really appreciate for respond to me. I am still looking forward to get reply back from you.

![Uploading Screenshot_2023-05-07-15-40-35-000_ru.iiec.pydroid3.jpg…]()

**Comment by rohan472000** [comment]
Try this - `pip install psycopg2-binary`

**Reply by doublyU** [reply]
It didn't work yet

**Comment by dvarrazzo** [comment]
I understand that Pydroid3 is a framework for Android. What platform would that be?

Most likely, psycopg2-binary will not work, and installing the library from source requires depending libraries compiled for the same platform (libpq, libssl in order to support encryption), or building them from source.

**Reply by doublyU** [reply]
I think so

**Comment by alslwyasad31-ui** [comment]
,  __main__.__dict__)
    ~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<string>", line 6
    <title>3D Racing Game - Speed Boost</title>
           ^

**Comment by th5831621-ux** [comment]
Hello

**Comment by thailandinifty-gif** [comment]
https://www.googletagmanager.com/ns.html?id=

**Reply by thailandinifty-gif** [reply]
https://github.com/psycopg/psycopg2/discussions/1572#discussioncomment-17607370
